appdev/connectors/mysql_connector.py

```diff
@@ -99,26 +107,53 @@ def scale_connection_pool(self, num_connections):
         connection.close()
     self.num_connections = num_connections
 
-  def read_batch(self, table, start, end, interval_size=10):
+  def read_batch(self, table, start=0, end=None, interval_size=10):
     input_queue = Queue()
     result_queue = Queue()
     a = start
+    if end is None:
+      end = self.get_count(table)
     while a <= end:
-      input_queue.put((a, a + interval_size))
+      b = a + interval_size
+      read_query = "SELECT * FROM {} LIMIT {} OFFSET {}" \
+                     .format(table, b-a+1, a)
+      input_queue.put(read_query)
       a += interval_size
     for _ in range(self.num_threads):
-      ReaderThread(self.connection_pool, table, input_queue,
-                   result_queue=result_queue).start()
+      ReaderThread(self.connection_pool, input_queue, result_queue).start()
     input_queue.join()
     rows = []
     while not result_queue.empty():
       rows.extend(result_queue.get())
     return rows
 
+  def get_count(self, table):
+    connection = self.connection_pool.get()
+    cursor = connection.cursor()
+    cursor.execute('SELECT COUNT(*) FROM {}'.format(table))
+    result = []
+    for row in cursor:
+      result.append(row)
+    self.connection_pool.put(connection)
+    return result[0][0]
```

Comment:
is this necessay if you are just doing result[0][0] why do a for loop for cursor?

Reply:
`cursor` will contain one row with one field, it'll be something like `[(2,)]`. I could do something else to only extract the first row, but it won't be any more efficient since there's only one row.